content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,861 @@
+`function_to_use()`. Here, we use it to build a separate contingency table of each
+variable in `food` crossed with illness status (`stegen$ill`); note that we show
+only the first few tables:
+
+```{r stegen_food_tables, echo = c(1,3), eval = -2}
+food_tables <- lapply(food, table, stegen$ill)
+food_tables
+head(food_tables)
+```
+
+
+### Realising multiple Chi-squared tests
+
+The same principle used to compute several contingency tables can be used to
+compute Chi-square tests for each table, using `lapply()`; note that we show
+only the first few tests:
+
+```{r stegen_food_tests, echo = c(1,3), eval = -2}
+food_tests <- lapply(food_tables, chisq.test)
+food_tests
+head(food_tests)
+```
+
+Looking at these results, it seems a large number of food items are
+significantly correlated to the illness. Which food item is the biggest suspect?
+We will try and address this question using risk ratios.
+
+
+
+## Risk ratios
+
+### For one variable 
+
+The [risk ratio](https://en.wikipedia.org/wiki/Risk_ratio) is defined as the
+ratio between the proportion of illness in one group (typically 'exposed') vs
+another ('non-exposed'). For instance, let us consider the contingency table of
+pork consumption and illness:
+
+```{r stegen_pork_table}
+food_tables$pork
+```
+
+This can be turned into proportions using `prop.table()`, specifying that we want
+proportions by rows (`margin = 1`):
+
+```{r stegen_pork_prop}
+pork_prop <- prop.table(food_tables$pork, margin = 1)
+pork_prop
+```
+
+the corresponding risk ratio is:
+
+```{r stegen_pork_rr}
+## ill & exposed: row 2, column 2
+pork_prop[2, 2] # row 2, column 2
+
+## ill & non-exposed: row 1, column 2
+pork_prop[1, 2]
+
+## compute the risk ratio
+pork_rr <- pork_prop[2, 2] / pork_prop[1, 2]
+pork_rr
+```
+
+Note the use of the syntax `[rows, columns]` to subset specific rows and columns
+of 2-dimensional objects.
+
+
+### Multiple variables
+
+In the example above, we have computed the risk ratio using a simple recipe:
+
+1. select a contingency table
+2. convert raw numbers to proportions (using `prop.table()`)
+3. isolating the risks in exposed and non-exposed groups, and compute their
+   ratio
+
+Repeating these tasks manually for all variables in `food` would be
+cumbersome. As an alternative, we can define the steps above as a function,
+i.e. a generic recipe which we can then apply to any contingency table. This new
+function will be called `risk_ratio()`, and is defined as follows:
+
+```{r stegen_risk_ratio_function}
+risk_ratio <- function(tab) { # 'tab' is a placeholder for the input 2x2 table
+  ## compute table of proportions
+  tab_prop <- prop.table(tab, margin = 1)
+  
+  ## compute the risk ratio
+  result <- tab_prop[2, 2] / tab_prop[1, 2]
+
+  ## return the output
+  return(result) # this will be the output of the function
+}
+```
+
+We can now try this function on any of the tables in `food_tables`, for instance:
+
+```{r rr_demo}
+food_tables$pork
+risk_ratio(food_tables$pork)
```

Comment:
chances of chucking in confidence intervals? Ppl not going to be happy with just riskratios,....